cmake/modules/FindMariaDBClient.cmake

```diff
@@ -66,7 +66,7 @@ if(NOT TARGET MariaDBClient::MariaDBClient)
     set_target_properties(MariaDBClient::MariaDBClient PROPERTIES
                                                        INTERFACE_INCLUDE_DIRECTORIES "${MARIADBCLIENT_INCLUDE_DIR}"
                                                        INTERFACE_COMPILE_DEFINITIONS HAS_MARIADB=1)
-    if(CORE_SYSTEM_NAME STREQUAL osx)
+    if(CORE_SYSTEM_NAME STREQUAL osx OR HOST MATCHES "steamdeck")
```

Comment:
I think you would be better off just disabling CLIENT_PLUGIN_AUTH_GSSAPI_CLIENT in the dependency build (as webos already does). No real reason it has to be on for any platform, and the way we build (static plugins) the pkg-config files dont report plugin requirements in the Libs output, so theres no easy way to get the requirements that are required due to the static build.

Reply:
> I think you would be better off just disabling CLIENT_PLUGIN_AUTH_GSSAPI_CLIENT

That's the approach I originally took before figuring out how to solve the missing library symbols, then I solved the missing symbols with this change, but I see your point and think I should go back to disabling `CLIENT_PLUGIN_AUTH_GSSAPI_CLIENT`.

Reply:
Does it pick up `gssapi_krb5` from somewhere else? The depends build should not build a shared library version of it. So this might get pulled in from a system installed mariadb connector.

Reply:
Not the slightest clue. It's unlikely that there's a system mariadb connector. The Steam Deck is based on a super minimal, stripped down fork of Arch. This shows the packages I had to install to get depends and Kodi to build: https://kodi.wiki/view/Kodi_Steam_Deck_development

Reply:
the real question is if kodi will ever have a need for kerberos auth on mysql/mariadb? I very much doubt that